### ISSUE TITLE
Rename bug for derived class method

### ISSUE THREAD

**Issue by realvictorprm** [mono/CppSharp#813]
##### Brief Description
The generated code of the CSharp tests is _uncompileable_ because one function signature isn't renamed to PascalCase. 

##### Further details
The part which is wrongly generated is this:
```
class DLL_API HasPropertyWithDerivedType
{
public:
    HasPropertyWithDerivedType();
    HasOverrideOfHasPropertyWithDerivedType* hasPropertyWithDerivedTypeSubclass;
    virtual void causeRenamingError();
};

class DLL_API HasOverrideOfHasPropertyWithDerivedType : public HasPropertyWithDerivedType
{
public:
    HasOverrideOfHasPropertyWithDerivedType();
    virtual void causeRenamingError();
};
```

The function `causeRenamingError` in class `HasOverrideOfHasPropertyWithDerivedType` is not renamed to PascalCase. Furthermore the base class function is spelled wrong too (should be PascalCase too). The base class function is renamed correctly.

PS: The PR #812 cannot be merged due to a renaming bug in the CSharpTests.
##### Used headers and code files

From the CSharpTest project:
CSharp.h
CSharp.cpp

##### Stack trace or incompilable generated code

```
    public unsafe partial class HasOverrideOfHasPropertyWithDerivedType : global::CSharp.HasPropertyWithDerivedType, IDisposable
    {
        public override void causeRenamingError()
        {
            base.causeRenamingError();
        }
        
    }

    public unsafe partial class HasPropertyWithDerivedType : IDisposable
    {
        public virtual void CauseRenamingError()
        {

        }
    }
```
(This reduced to the important code)

**Comment by realvictorprm** [comment]
should be fixed with pr #812

**Comment by realvictorprm** [comment]
Fixed
